lib/improver/tests/utilities/cube_manipulation_new/test_MergeCubes.py

```diff
@@ -0,0 +1,406 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2019 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""
+Unit tests for the "cube_manipulation.MergeCubes" plugin.
+"""
+
+import unittest
+import numpy as np
+from datetime import datetime as dt
+
+import iris
+from iris.exceptions import DuplicateDataError, MergeError
+from iris.tests import IrisTest
+
+from improver.utilities.cube_checker import find_threshold_coordinate
+from improver.utilities.cube_manipulation_new import MergeCubes
+from improver.utilities.warnings_handler import ManageWarnings
+from improver.tests.set_up_test_cubes import (
+    set_up_variable_cube, set_up_probability_cube)
+
+
+class Test__init__(IrisTest):
+    """Test the __init__ method"""
+
+    def test_basic(self):
+        """Test default parameters"""
+        plugin = MergeCubes()
+        self.assertSequenceEqual(plugin.silent_attributes,
+                                 ["history", "title", "mosg__grid_version"])
+        self.assertSequenceEqual(plugin.coord_mismatch_error_keys,
+                                 ["threshold"])
+
+
+class Test__equalise_cubes(IrisTest):
```

Comment:
`_equalise_cubes` is an odd function because it just calls a series of functions that are unit tested elsewhere in the module.  So I didn't think too many checks were needed, and I chose the aspects for which input cubes were easiest to set up (`_equalise_coords` requires a particularly convoluted setup).  I should cover the float64 aspect though.

Reply:
I don't think you've done the float64 unit test yet.

Reply:
Now added a couple of datatype tests covering demotion of float64 and non-promotion of integer types.